clippy_lints/src/utils/mod.rs

```diff
@@ -602,6 +602,37 @@ pub fn contains_return(expr: &hir::Expr<'_>) -> bool {
     visitor.found
 }
 
+struct FindMacroCalls<'a> {
+    names: Vec<&'a str>,
+    result: Vec<Span>,
+}
+
+impl<'a, 'tcx> Visitor<'tcx> for FindMacroCalls<'a> {
+    type Map = Map<'tcx>;
+
+    fn visit_expr(&mut self, expr: &'tcx Expr<'_>) {
+        if self.names.iter().any(|fun| is_expn_of(expr.span, fun).is_some()) {
+            self.result.push(expr.span);
+        }
+        // and check sub-expressions
+        intravisit::walk_expr(self, expr);
+    }
+
+    fn nested_visit_map(&mut self) -> NestedVisitorMap<Self::Map> {
+        NestedVisitorMap::None
+    }
+}
+
+/// Finds calls of the specified macros in a function body.
+pub fn find_macro_calls(names: Vec<&str>, body: &'tcx Body<'tcx>) -> Vec<Span> {
```

Comment:
```suggestion
pub fn find_macro_calls(names: &[&str], body: &Body<'_>) -> Vec<Span> {
```
We can be more general in what we accept by taking a slice.

NIT about the lifetime name, we can elide it (I was puzzled for a while because the lifetime was not declared in the generic arguments, later found it's because of `#[feature(in_band_lifetimes)]` :smile:)

Reply:
So this is how it's done!
Thanks, I gave up and just used `Vec` after about an hour of wrestling against the compiler :grin: